src/base/vec_storage.rs

```diff
@@ -410,3 +411,21 @@ impl<T> Extend<T> for VecStorage<T, Dynamic, U1> {
         self.nrows = Dynamic::new(self.data.len());
     }
 }
+
+impl<T> DMatrix<T> {
+    /// Creates a new heap-allocated matrix from the given [VecStorage].
+    pub const fn from_vec_storage(storage: VecStorage<T, Dynamic, Dynamic>) -> Self {
+        // This is sound because the dimensions of the matrix and the storage are guaranteed
+        // to be the same
+        unsafe { Self::from_data_statically_unchecked(storage) }
+    }
+}
+
+impl<T> DVector<T> {
+    /// Creates a new heap-allocated matrix from the given [VecStorage].
+    pub const fn from_vec_storage(storage: VecStorage<T, Dynamic, U1>) -> Self {
+        // This is sound because the dimensions of the matrix and the storage are guaranteed
+        // to be the same
+        unsafe { Self::from_data_statically_unchecked(storage) }
+    }
+}
```

Comment:
I would suggest to put this on `matrix.rs`, bellow the impl block containing `from_data_statically_unchecked`.
Perhaps we should add a `TODO` comment saying that this should be removed in favor of `Matrix::from_data` once `const-fn` support generic trait bounds.